instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/internal/AwsXrayEnvSpanLinksExtractor.java

```diff
@@ -42,22 +42,32 @@ public void extract(
   }
 
   public static void extract(SpanLinksBuilder spanLinks) {
-    String parentTraceHeader = System.getenv(AWS_TRACE_HEADER_ENV_KEY);
-    if (parentTraceHeader == null || parentTraceHeader.isEmpty()) {
+    Map<String, String> contextMap = getSystemPropertyOrEnvironmentVariable();
+    if (contextMap.isEmpty()) {
       return;
     }
     Context xrayContext =
-        AwsXrayPropagator.getInstance()
-            .extract(
-                Context.root(),
-                Collections.singletonMap(AWS_TRACE_HEADER_PROPAGATOR_KEY, parentTraceHeader),
-                MapGetter.INSTANCE);
+        AwsXrayPropagator.getInstance().extract(Context.root(), contextMap, MapGetter.INSTANCE);
     SpanContext envVarSpanCtx = Span.fromContext(xrayContext).getSpanContext();
     if (envVarSpanCtx.isValid()) {
       spanLinks.addLink(envVarSpanCtx, LINK_ATTRIBUTES);
     }
   }
 
+  private static Map<String, String> getSystemPropertyOrEnvironmentVariable() {
+    String traceHeader = System.getProperty(AWS_TRACE_HEADER_PROP);
```

Comment:
Is it ok to assume that any non-null/empty value will be a valid header or should it attempt to parse both in case the property version is invalid?

Reply:
I see this PR as a transition from the env variable to a system property. We still do want to continue support env variable for legacy code, but the system property should have more priority.
```
if system property is set
   read traceHeader from it
else if env variable is set
   read traceHeader from it
```

Reply:
Yeah, I understand that.  Just wondering if anything special should be done if they're both set, but the system property is corrupt for some reason.

Reply:
I think we should fail fast in this case. Adding a fallback path can be something unnoticeable and surprising in the end for end users

Reply:
How do we identify something as corrupt? Something other than the key we expect? Also what do we expect here in terms of "fail fast"?